### PR TITLE
Implement Reports API

### PR DESCRIPTION
**Pull request by XaosSintez**

### CODE REVIEW

File: src/Entities/Schema.php [hunk 1]
```diff
@@ -195,6 +202,36 @@ public function __construct()
             '3dsecure/{3dsecureId}' => function (array $content) {
                 return new ThreeDSecure($content);
             },
+            'reporting-currencies' => function (array $content) {
```
Comment (slavcodev) [inline src/Entities/Schema.php]: This seems a bad conflict resolving. I believe we remove this endpoint and resource.

File: src/Entities/Reports/DisputesReport.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * ```json
+ * {
+ *   "data"
+ * }
+ * ```
+ *
+ * @author Arman Tuyakbayev <arman.tuyakbayev@rebilly.com>
+ * @version 0.1
+ */
+final class DisputesReport extends Entity
```
Comment (slavcodev) [inline src/Entities/Reports/DisputesReport.php]: All of these classes seems should not extends `Entity` once they do not have `ID`. The `Resource` class would be better, the parent for `Entity`.

File: src/Entities/Schema.php (hunk 1 above)
Comment (XaosSintez) [inline src/Entities/Schema.php]: Yes, it seems so. Fixed

File: src/Entities/Reports/DisputesReport.php (hunk 2 above)
Comment (XaosSintez) [inline src/Entities/Reports/DisputesReport.php]: Yes, i thought about it and forgot in the end. Fixed, thanks

**Pull request closed without merging**